core/network/fabric-interop-cc/contracts/interop/membership.go

```diff
@@ -58,16 +347,157 @@ func (s *SmartContract) CreateMembership(ctx contractapi.TransactionContextInter
 }
 
 // UpdateMembership cc is used to update an existing Membership in the ledger
-func (s *SmartContract) UpdateMembership(ctx contractapi.TransactionContextInterface, membershipJSON string) error {
+// TODO: Remove call to 'updateMembership' after creating Corda IIN Agents.
+func (s *SmartContract) UpdateMembership(ctx contractapi.TransactionContextInterface, counterAttestedMembershipSerialized string) error {
+	// Check if the caller has IIN agent privileges
+	if isIINAgent, err := isClientIINAgent(ctx); err != nil {
+		return fmt.Errorf("IIN Agent client check error: %s", err)
+	} else if !isIINAgent {
+		// Check if the caller has network admin privileges
+		if isAdmin, err := isClientNetworkAdmin(ctx); err != nil {
+			return fmt.Errorf("Admin client check error: %s", err)
+		} else if !isAdmin {
+			return fmt.Errorf("Caller neither a network admin nor an IIN Agent; access denied")
+		}
+		return s.updateMembership(ctx, counterAttestedMembershipSerialized)		// HACK to handle unattested memberships (for Corda) for backward compatibility
+	}
+
+	// Check if caller is one of the authorized IIN agents
+	membershipLocal, err := s.GetMembershipBySecurityDomain(ctx, membershipLocalSecurityDomain)
+	if err != nil {
+		return err
+	}
+	membershipLocalDecoded, err := decodeMembership([]byte(membershipLocal))
+	if err != nil {
+		return fmt.Errorf("Unmarshal error: %s", err)
+	}
+	callerId := ctx.GetClientIdentity()
+	callerCert, err := callerId.GetX509Certificate()
+	if err != nil {
+		return err
+	}
+	err = checkUserInMembership(callerCert, membershipLocalDecoded)
```

Comment:
Same comments as in `CreateMembership`. Also one more thing, seems like this is exact same code as in `Create`, can you move this to another helper function so as to remove code duplication?

Reply:
Fixed as above.